hs/src/Reach/Eval/Core.hs

```diff
@@ -2432,6 +2456,15 @@ doBalanceUpdate mtok op = \case
     let up_rator = SLV_Prim $ SLPrim_PrimDelay at (SLPrim_op op) [] [(Public, rhs)]
     -- Find the current balance of mtok
     bsv <- getBalanceOf mtok
+    -- Assume we can pay into contract
+    when (op == S_ADD) $ do
```

Comment:
We can also assume that subtraction works, because we know we're already going to check that the balance is sufficient for the transfer